lib/Solver/SolverCmdLine.cpp

```diff
@@ -112,17 +112,19 @@ cl::opt<bool> UseAssignmentValidatingSolver(
     cl::desc("Debug the correctness of generated assignments (default=false)"),
     cl::cat(SolvingCat));
 
-
-void KCommandLine::HideOptions(llvm::cl::OptionCategory &Category) {
+void KCommandLine::KeepOnlyCategories(
+    std::set<llvm::cl::OptionCategory *> categories) {
```

Comment:
This function should not take a copy of the category set, as it only needs to inspect the contents
```suggestion
    std::set<llvm::cl::OptionCategory *> const& categories) {
```